django_lti_tool_provider/views.py

```diff
@@ -45,10 +46,19 @@ def post(self, request, *args, **kwargs):
         return self.process_request(request)
 
     def process_request(self, request):
+        if request.user.is_authenticated():
+            try:
+                lti_parameters = self._get_lti_parameters_from_request(request)
+                if not self._right_user(request.user, lti_parameters):
+                    _logger.debug(u"Logging out user %s in favor of new LTI session.", request.user.username)
+                    logout(request)
+            except (oauth2.Error, AttributeError):
+                # Not a new visit, or better to keep existing auth.
+                pass
```

Comment:
@Kelketek this function is not called than, as requests does not go through LTI handler. Basically, LTI request is sent once per "session", i.e. when user get to actual app LTI already done what it is supposed to do and app works as if there were no LTI at all, i.e. requests go to ordinary URLs, handled by normal views, etc.

Your answer still not addresses logging exceptions part :)